Mage/src/main/java/mage/abilities/AbilityImpl.java

```diff
@@ -1328,6 +1326,26 @@ public void setSourceObjectZoneChangeCounter(int sourceObjectZoneChangeCounter)
         this.sourceObjectZoneChangeCounter = sourceObjectZoneChangeCounter;
     }
 
+    @Override
+    public void updateSourceObjectZoneChangeCounter(Game game, boolean force) {
+        if (!(this instanceof MageSingleton) && (force || sourceObjectZoneChangeCounter == 0 )) {
+            setSourceObjectZoneChangeCounter(getCurrentSourceZCC(game));
```

Comment:
That's force param smells bad. Need research on usage and docs for it. Maybe it can be renamed to "copiedAbility" or like that (or maybe not). Maybe it can be removed at all (or maybe not).

Also `updateSourceObjectZoneChangeCounter` must be renamed to `initSourceObjectZoneChangeCounter`

Reply:
It definitely can't be entirely removed without additional work, I tried that just to see and there were multiple test failures. I'll investigate further.